Source/DataSources/KmlDataSource.js

```diff
@@ -279,9 +279,14 @@ define([
         return url;
     }
 
-    function getOrCreateEntity(node, entityCollection) {
+    // an optional context is passed to allow for some malformed kmls (thos with multiple geometries with same ids) to still parse
+    // correctly, as they do in Google Earth.
+    function getOrCreateEntity(node, entityCollection, context) {
         var id = queryStringAttribute(node, 'id');
-        id = defined(id) ? id : createGuid();
+        id = defined(id) && id !=="" ? id : createGuid();
```

Comment:
Use single quotes for strings. Here you may just want to use `id.length !== 0` since you are just checking for a non-empty string.